consensus/ethash/sealer.go

```diff
@@ -150,3 +164,122 @@ search:
 	// during sealing so it's not unmapped while being read.
 	runtime.KeepAlive(dataset)
 }
+
+// remote starts a standalone goroutine to handle remote mining related stuff.
+func (ethash *Ethash) remote(resultCh chan *types.Block) {
```

Comment:
Why does this take a channel as a parameter? I saw that sometimes you pass in `nil`, but wouldn't that be the same as `ethash.resultCh` being nil?

Reply:
Do we even want to start this goroutine in the first place if there's no result channel?